dev/com.ibm.ws.concurrent/src/com/ibm/ws/concurrent/internal/ScheduledTask.java

```diff
@@ -796,6 +824,8 @@ public boolean cancel(boolean mayInterruptIfRunning) {
 
                     canceled = true;
                 } else {
+                    // Let pending work (such as reschedule) on other threads go first, and then refresh status
+                    Thread.yield();
```

Comment:
Interesting.  I would have never thought to use yield.
Is there any concern that yielding here could cause us to continually check the status while another thread is rescheduling the task?

Reply:
Yielding will have the opposite effect. It will make that less likely by always encouraging other threads to go first.